src/server/cache.rs

```diff
@@ -71,14 +78,25 @@ async fn tile_cache_updater(state: &ServerState) {
         tiles_cache,
         reqwest_client,
         adm_endpoint_url,
-        metrics,
+        settings,
         ..
     } = state;
 
     trace!("tile_cache_updater..");
-    let keys: Vec<_> = tiles_cache.read().await.keys().cloned().collect();
+    let tiles = tiles_cache.read().await;
```

Comment:
I'm a little concerned that the cache read lock (guard) is now held throughout this function and across await points, we should probably avoid that if we can.

Reply:
Moved the cleanup routine into it's own block and have the locks more isolated.